include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -415,18 +415,13 @@ class transform_iterator
     typedef decltype(__my_unary_func_(::std::declval<typename ::std::iterator_traits<_Iter>::reference>())) reference;
     typedef typename ::std::remove_reference<reference>::type value_type;
     typedef typename ::std::iterator_traits<_Iter>::pointer pointer;
-    typedef typename ::std::random_access_iterator_tag iterator_category;
+    typedef typename ::std::iterator_traits<_Iter>::iterator_category iterator_category;
 
-    transform_iterator(_Iter __it, _UnaryFunc __unary_func) : __my_it_(__it), __my_unary_func_(__unary_func)
-    {
-        static_assert((::std::is_same<typename ::std::iterator_traits<_Iter>::iterator_category,
-                                      ::std::random_access_iterator_tag>::value),
-                      "Random access iterator required.");
-    }
-    transform_iterator(const transform_iterator& __input)
-        : __my_it_(__input.__my_it_), __my_unary_func_(__input.__my_unary_func_)
+    transform_iterator(_Iter __it = _Iter(), _UnaryFunc __unary_func = _UnaryFunc())
```

Comment:
where was separate `transform_iterator() = default` lost? I thought we agreed on that on and on 
`transform_iterator(_Iter __it, _UnaryFunc __unary_func = _UnaryFunc())`

Reply:
We can not add `transform_iterator() = default` because in this case we should add explicit initializer to `my_unary_function` and it does not make transform_iterator trivially default constructible. Another way - is to erase `const` qualifier for `my_unary_function`, but I believe it is bad solution, so I think we prefer not to change implementation.